src/singletons/NativeMessaging.cpp

```diff
@@ -158,9 +205,9 @@ namespace nm::client {
 
 // SERVER
 NativeMessagingServer::NativeMessagingServer()
-    : thread(*this)
+    : thread(new ReceiverThread(*this))
 {
-    this->thread.setObjectName("NativeMessagingReceiver");
+    this->thread->setObjectName("NativeMessagingReceiver");
```

Comment:
Rename the thread so it fits within posix thread limitations
```suggestion
    this->thread->setObjectName("C2NMReceiver");
```